src/main/scala/com/stratio/receiver/RabbitMQInputDStream.scala

```diff
@@ -119,28 +119,38 @@ class RabbitMQReceiver(params: Map[String, String], storageLevel: StorageLevel)
   }
 
   def getQueueName(channel: Channel): String = {
+    // Get the queue name to use (explicit vs auto-generated).
     val queueName = checkQueueName()
-    routingKeys match {
-      case Some(routingKey) => {
-        log.info("declaring topic queue")
-        channel.exchangeDeclare(exchangeName, exchangeType, true)
 
-          val queue = channel.queueDeclare(queueName, true, false, false, new util.HashMap(0)).getQueue()
-
-        for (routingKey: String <- routingKey.split(",")) {
-          log.info("binding to routing key " + routingKey)
-          channel.queueBind(queue, exchangeName, routingKey)
+    // If no exchange was specified, then create a direct connection to the queue.
+    if(exchangeName == null || exchangeName.trim.isEmpty) {
```

Comment:
Thank you for the suggested change; I have made this update, and all works expected after testing again.